Support both DataFrames and NumPy arrays

mario-koddenbrock opened the pull request:
Refactored tensor preparation by introducing a `to_numpy` helper function, which ensures compatibility with both pandas DataFrames/Series and NumPy arrays. This makes the codebase more flexible and robust for different input types.

mario-koddenbrock committed:
Support both DataFrames and NumPy arrays in tensor conversion

Refactored tensor preparation by introducing a `to_numpy` helper function,
which ensures compatibility with both pandas DataFrames/Series and NumPy arrays.
This makes the codebase more flexible and robust for different input types.

LennartPurucker commented:
Heyho Mario, thank you for the contribution!

Mhm, this implementation is currently in conflict with the interface. The interface expects the data to be in pandas format (https://github.com/LennartPurucker/finetune_tabpfn_v2/blob/master/finetuning_scripts/finetune_tabpfn_main.py#L62-L63).
Your code changes only adapt some parts where the data might be used, and other parts of the code might still require a dataframe (e.g., for plotting) and would crash with the current logic.

To support numpy input, I instead recommend adapting the expected input type to be pandas and numpy, and adding code at the start of the function that transforms any input we get into a dataframe. This way, the code will keep working as intended, and numpy is supported. Could you re-implement your PR in such a way? 

In general, I recommend to always keep/work with a dataframe and only switch to numpy when needed.

LennartPurucker reviewed (changes requested):
See comment above.

mario-koddenbrock committed:
Ensure consistent DataFrame input format in `fine_tune_tabpfn`

Coerced input arrays to pandas DataFrames/Series at the start of the function,
ensuring downstream compatibility and preserving the existing pandas-based interface.
This allows support for both NumPy and pandas inputs while avoiding partial tensor conversion issues.

mario-koddenbrock committed:
Fix formatting issue in data_utils.py by removing unnecessary blank line

mario-koddenbrock commented:
Heyho, thanks a lot for the helpful feedback!

You're absolutely right — I initially based my understanding on the argument documentation, which already mentioned support for both pd.DataFrame/pd.Series and np.ndarray inputs. At the time, I overlooked that the validate_tabpfn interface (and other parts like plotting) still require the data to be in pandas format internally.

I've now updated the implementation as you suggested:

All inputs (X_train, y_train, X_val, y_val) are explicitly coerced to pandas objects at the start of the fine_tune_tabpfn function.

LennartPurucker commented on finetuning_scripts/finetune_tabpfn_main.py:
```diff
@@ -23,7 +23,7 @@
 from finetuning_scripts.metric_utils.ag_metrics import get_metric
 from finetuning_scripts.training_utils.ag_early_stopping import AdaptiveES
-from finetuning_scripts.training_utils.data_utils import get_data_loader
+from finetuning_scripts.training_utils.data_utils import get_data_loader, to_numpy
```
```suggestion
from finetuning_scripts.training_utils.data_utils import get_data_loader
```

LennartPurucker reviewed (approved):
LGTM!

LennartPurucker committed:
Update finetuning_scripts/finetune_tabpfn_main.py

LennartPurucker commented:
Awesome, thank you very much! :)

LennartPurucker commented:
> pd.DataFrame/pd.Series and np.ndarray inputs

Indeed, so it seems I was overly optimistic with the documentation :D
